### PR TITLE
Convert `pj` to RzShell

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Moves the `pj` command to RzShell. It also does some clean ups:

- Removes `pj.`, because it was an alias for `pj @ 0`
- Removes `pj..`, because the JSON path function always returned `NULL` (never a valid value), and did not document what it does in the first place.
- Print message for user if no valid data was found at address.
- Change description of command to be more clear it only prints formatted JSON strings.
- Removes unused `size/len` argument.
- Adds a new test for `pj`.
- Adds some general documentation how to implement or find a command.

**Test plan**

Added

**Closing issues**

None, but checks a box in https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -456,3 +456,18 @@ int main() {
 - You should use these modifiers consistently in both function definition and declaration.
 - You should use these modifiers when and only when it makes sense. For example, if your function returns `const char *`, the caller should not free it because of the `const`. So specifying `RZ_BORROW` in this case is probably redundant.
 - Since they are used as indications to developers with no special compiler-time restrictions, there is no good way to check if you have used them correctly.
+
+# Useful utilities
+
+## Finding command handler
+
+If you need to find the logic for a given command run `./sys/rzshell_which.py <command>` to get the
```
Comment (XVilka) [inline DEVELOPERS.md]: I think it was already described in `doc/rzshell.md`. Maybe better just refer to it instead?

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -3517,6 +3474,19 @@ RZ_IPI RzCmdStatus rz_cmd_disassembly_function_handler(RzCore *core, int argc, c
 	return RZ_CMD_STATUS_OK;
 }
 
+RZ_IPI RzCmdStatus rz_print_current_block_json_handler(RzCore *core, int argc, const char **argv) {
+	rz_return_val_if_fail(core, RZ_CMD_STATUS_ERROR);
+	bool enable_color = rz_config_get_i(core->config, "escr.color") != 0;
+	char *res = rz_print_json_indent((const char *)core->block, enable_color, "  ", NULL);
+	if (RZ_STR_ISEMPTY(res)) {
+		rz_cons_printf("Couldn't find a JSON string.\n");
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: Shouldn't this return RZ_CMD_STATUS_ERROR?

Comment (XVilka) [reply librz/core/cmd/cmd_print.c]: Also maybe better using `RZ_LOG_ERROR()` here?

File: librz/core/cmd_descs/cmd_print.yaml [hunk 3]
```diff
@@ -864,6 +864,10 @@ commands:
         summary: Print all instructions at the current function
         cname: print_instructions_function
         args: []
+  - name: pj
+    summary: Print formatted JSON at current offset.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_print.yaml]: Maybe expand the description to note that it tries to parse it first?

**Review by XVilka — CHANGES_REQUESTED** [review]
Code is fine, with mostly documentation nitpicks

File: librz/core/cmd/cmd_print.c (hunk 2 above)
Comment (wargio) [reply librz/core/cmd/cmd_print.c]: yeah, throw an error saying invalid json (empty string).

File: DEVELOPERS.md (hunk 1 above)
Comment (Rot127) [reply DEVELOPERS.md]: Removed it. I can't tell for the sake of it why I didn't have the idea to check `doc/` first.__

**Review by XVilka — APPROVED** [review]

**Pull request merged**